cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -556,21 +579,20 @@ private predicate simpleInstructionLocalFlowStep(Instruction iFrom, Instruction
   // reassignment of the parameter indirection, including a conditional one that
   // leads to a phi node.
   exists(InitializeIndirectionInstruction init |
-    iFrom = init and
-    iTo.(LoadInstruction).getSourceValueOperand().getAnyDef() = init and
+    opFrom.getAnyDef() = init and
+    iTo.(LoadInstruction).getSourceValueOperand() = opFrom and
     // Check that the types match. Otherwise we can get flow from an object to
     // its fields, which leads to field conflation when there's flow from other
     // fields to the object elsewhere.
-    init.getParameter().getType().getUnspecifiedType().(DerivedType).getBaseType() =
-      iTo.getResultType().getUnspecifiedType()
+    initializeParameterOfType(init, iTo.getResultType())
   )
```

Comment:
I hope this rule is no longer needed. There's a rule in `simpleOperandLocalFlowStep` to propagate flow to the operand of the `Load`, and there's a rule in this predicate to propagate flow from `getSourceValueOperand` of any `CopyInstruction` to the result of that instruction. The comment can be moved to `simpleOperandLocalFlowStep`.

Reply:
Good catch. I've removed it in 9887d8b7ebe2cee6458fbf27a3c84f2976934436.